testdata/p4_16_samples/psa-i2e-cloning-basic-bmv2.stf

```diff
@@ -0,0 +1,19 @@
+
+mc_mgrp_create 3
+mc_node_create 0 6
+mc_node_create 0 7
+mc_node_create 0 8
+mc_node_associate 3 0
+mc_node_associate 3 1
+mc_node_associate 3 2
+
+packet 1 000000000002 000000000000 ffff
```

Comment:
I believe Antonin made similar comments on the corresponding changes you propose in the p4lang/behavioral-model PR to add cloning support for PSA, but the clone session ids should not directly use multicast group ids.  That is, multicast group id 3 and clone session id 3 refer to different configurable 'objects' in the switch, and clone session id 3 should not use the multicast group 3 you are using the 'mc_' commands above to configure.

It would be a good idea to add support for some or perhaps all of the 'mirroring_' commands in simple_switch_CLI, probably copied and pasted into psa_switch_CLI, that are used for configuring mirroring sessions, where mirroring is another name for cloning in the P4 world.  https://github.com/p4lang/behavioral-model/blob/master/docs/runtime_CLI.md#mirroring_add-mirroring_add_mc-mirroring_delete-mirroring_get

I can probably find time to help add support for those commands to the STF files, since I added the corresponding support for the 'mc_' commands in STF files a couple of years ago, and the changes for this should be very similar.  The addition of the 'mc_' command support in STF files was added with this commit: https://github.com/p4lang/p4c/commit/7a8fae

Looking at that commit, it adds support for the 'mirroring_add' command, too, but not 'mirroring_add_mc', which would be useful to add for STF testing purposes, and looks like it is probably a one-line change to the bmv2stf.py program.

Reply:
@jafingerhut
With the latest commit, I added support in pswitch_CLI and bmv2stf for those mirroring commands and fixed up this STF test to not mistake multicast group ids for clone session ids. Please review when you get a chance.